docs/adr/0018-audit-log-archive.md

```diff
@@ -0,0 +1,43 @@
+# ADR: Archiving Growing Audit Logs
+
+## Status
+Proposed
+
+## Context
+We need to determine the most suitable approach for archiving growing audit logs. The data retention period specified by OCIO is 6 years. While we plan to leave the TTA Hub data used for analytics intact, the audit logs providing a historic view of how that data changed, need to be managed. The main reason behind the need to implement an archiving strategy is to ensure the rapidly growing data doesn't stand in the way of the application performance. We have identified two alternatives for archiving the audit logs:
+
+1. Postpone Archiving: The first option is to postpone archiving until the end of the 6-year data retention period. This means that no specific archiving mechanism will be implemented until that time.
+
+2. Quarterly Archiving using Node.js cron and JavaScript/TypeScript: The second option is to implement a quarterly archiving process using Node.js cron functionality with JavaScript/TypeScript code. The code will retrieve the data from Postgres and save it to an S3 bucket. The S3 bucket already has server-side encryption enabled.
+
+### Advantages and disadvantages of option 1:
+By not implementing anything now, we don't incur costs of implementing the archiving, testing for accuracy, and the development of an admin UI to allow for seamless restores as needed. However, by allowing the audit logs to grow we are running a risk of a general application response time slowdown as well as significantly increased times to backup the database.
+
+### Advantages and disadvantages of option 2:
+The main advantage of implementing an archiving strategy now is that our application performance will not be impacted by rapidly growing audit log data. The production backups will operate as they do now without unnecessary slowdowns. A disadvantage of this option is that we will need to dedicate resources to implement, test and monitor the solution.
+
+
+## Decision
+After evaluating the advantages and disadvantages of both alternatives, we have decided to proceed with the second option: Quarterly Archiving to S3 using Node.js cron.
```

Comment:
@GarrettEHill Thanks for the detailed feedback! 👍 

> When was this decided, who was involved in making this decision?

The aim of this ADR was to present 2 options - archive now or postpone. It looks like as a team we are aligned on the need to archive now. I did modify the wording slightly to allow us more flexibility as far as how often to archive. We want to present a high level design to the IPT and gain their approval. I am hoping they will agree with our approach and let us implement the solution. I  created #1570 and incorporated the feedback from here to outline the different methods of the implementation. This would be a good start (we might need to adjust based on the IPT input).